### PR TITLE
Add 'owners-label' plugin to Prow to replace 'owner-label' and 'path-label' mungers.

### LINKED ISSUE #6033: Migrate owner-label and path-label mungers out of mungegithub
This is being used in kubernetes/community, now that we've migrated OWNERS handling to prow this should be relatively easy to re-implement as a plugin, or maybe add'l functionality for the existing label plugin?

/area prow
/area mungegithub

xref #6019
xref #3331

### PR DESCRIPTION
**Pull request by cjwagner**
The functionality of the `path-label` munger (regexp based path labeling) can now be achieved with OWNERS files thanks to #6411 so this is really just a migration of the `owner-label` munger.

fixes #6033 
/area prow
/bark
/cc @rmmh @BenTheElder

### CODE REVIEW

File: prow/plugins/owners-label/owners-label.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+const (
+	pluginName = "owners-label"
+)
+
+func init() {
+	plugins.RegisterPullRequestHandler(pluginName, handlePullRequest, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []string) (*pluginhelp.PluginHelp, error) {
+
```
Comment (kargakis) [inline prow/plugins/owners-label/owners-label.go]: Are you going to fix help in this PR? If not, you need at least a return statement.

File: prow/plugins/owners-label/owners-label_test.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+limitations under the License.
+*/
+
+package ownerslabel
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/Sirupsen/logrus"
```
Comment (kargakis) [inline prow/plugins/owners-label/owners-label_test.go]: We use the lowercased sirupsen import now.

File: prow/plugins/owners-label/owners-label_test.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	"k8s.io/gengo/examples/set-gen/sets"
```
Comment (kargakis) [inline prow/plugins/owners-label/owners-label_test.go]: We don't vendor this. You probably want the apimachinery package.

File: prow/plugins/owners-label/owners-label.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+*/
+
+package ownerslabel
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/util/sets"
```
Comment (kargakis) [inline prow/plugins/owners-label/owners-label.go]: nit: The way we group imports is incosistent across our codebase. We should come up with a convention, document, and follow.

**Comment by cjwagner** [comment]
PTAL, this PR should be more complete now.

**Pull request merged**